src/main/java/com/pro/sky/course2/coursework2/repository/JavaQuestionRepository.java

```diff
@@ -0,0 +1,47 @@
+package com.pro.sky.course2.coursework2.repository;
+
+import com.pro.sky.course2.coursework2.domain.Question;
+import com.pro.sky.course2.coursework2.exception.QuestinAlreadyAddedException;
+import com.pro.sky.course2.coursework2.exception.QuestionNotFoundException;
+import jakarta.annotation.PostConstruct;
+import org.springframework.stereotype.Repository;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+@Repository
+public class JavaQuestionRepository implements QuestionRepository{
+
+    private Set<Question> questions = new HashSet<>();
```

Comment:
```suggestion
    private final Set<Question> questions = new HashSet<>();
```
если можно делать final поля - лучше делать

Reply:
с модификатором final у меня вечная проблема - уже не одну статью прочитал и всё равно до конца не понмаю когда нужно/можно/зачем его использовать)

Reply:
примерно как с private - для безопасности) когда - везде, где можно его поставить лучше ставить (для полей)